### PR TITLE
added contributing file to docs folder

### PR DESCRIPTION
**Pull request by blueQcloud**

### CODE REVIEW

**Comment by wenzowski** [comment]
@matthieu-foucault we could set [post-checkout](https://git-scm.com/docs/githooks#_post_checkout) to `git submodule update --remote --init` but since we're managing all other dependencies from `make watch` maybe that's a better place?

I think the bigger process consideration here is probably using `git add -p` or similar UI to review changes before publishing.

**Comment by matthieu-foucault** [comment]
> @matthieu-foucault we could set [post-checkout](https://git-scm.com/docs/githooks#_post_checkout) to `git submodule update --remote --init` but since we're managing all other dependencies from `make watch` maybe that's a better place?

I'd do a post-checkout hook, without the `--remote` flag (otherwise that will update the submodule as soon as a new commit is available there, and we don't necessarily want that)

**Comment by wenzowski** [comment]
What happens with `git submodule update --init` if there's a dirty working directory? Do we clobber local changes?

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<h2>How to Contribute:</h2>
+
+CAS CIIP portal is accepting contributions via Github pull requests. (If you're new to GitHub, you might start with a [basic tutorial](https://help.github.com/en/github/getting-started-with-github/set-up-git) and check out a more detailed guide to [pull requests](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests))This document outlines the contribution work flow, contact points for developers and other resources to make getting your contribution accepted.
+
+<h2>Certificate of Origin</h2>
+
+By contributing to this project you agree to the [Certificate of Origin (DCO)](https://github.com/deis/deis/blob/master/DCODeveloper). This document was created by the Linux Kernel community and is a simple statement that you, as a contributor, have the legal right to make the contribution.
```
Comment (wenzowski) [inline docs/CONTRIBUTING.md]: If we're going to have a Certificate of Origin, we need that in this repo

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<h2>How to Contribute:</h2>
```
Comment (wenzowski) [inline docs/CONTRIBUTING.md]: I don't think we need embedded HTML here. Let's use native markdown? https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet

File: docs/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+<h2>Certificate of Origin</h2>
+
+By contributing to this project you agree to the [Certificate of Origin (DCO)](https://github.com/deis/deis/blob/master/DCODeveloper). This document was created by the Linux Kernel community and is a simple statement that you, as a contributor, have the legal right to make the contribution.
+
+<h2>Support Channels</h2>
+
+Before opening a new issue, it's helpful to search the project - it's likely that another user has already reported the issue you're facing, or it's a known issue that we're already aware of.
+[Create An Issue](https://github.com/bcgov/BC-Policy-Framework-For-GitHub/issues/new?title=Devhub%20Issue%3A%20Sample%20Contributing%20File%20%5Bshort%20description%20here%5D&body=%3E%20path%3A%20(do%20not%20delete)%20https%3A%2F%2Fgithub.com%2Fbcgov%2FBC-Policy-Framework-For-GitHub%2Fblob%2Fmaster%2FBC-Gov-Org-HowTo%2FSAMPLE-CONTRIBUTING.md%0A%20%3E%20(do%20not%20delete)%20devhub%20page%3A%20%2FSample-Contributing-File%0A%0A%23%23%20Devhub%20Content%20Issue%0A)
```
Comment (wenzowski) [inline docs/CONTRIBUTING.md]: this looks like it's creating an issue on https://github.com/bcgov/BC-Policy-Framework-For-GitHub/issues ...did you mean to do that?

File: docs/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+<h3>Getting Started</h3>
+*Fork the repository on GitHub
+*Read the [documentation](https://github.com/bcgov/cas-ciip-portal/blob/develop/documentation/developer-guide.md) for build instructions
+
+<h3>Contribution Flow</h3>
+*Create a topic branch from where you want to base your work.
+*Make commits of logical units.
+*Push your changes to a topic branch in your fork of the repository.
+*Submit a pull request
```
Comment (wenzowski) [inline docs/CONTRIBUTING.md]: markdown requires spaces for bullets to work

**Comment by wenzowski** [comment]
@blueQcloud a [quick preview](https://github.com/bcgov/cas-ciip-portal/pull/117/files?short_path=8595297#diff-8595297407c24b9ccb87b2a698cca9c2) shows there are now a bunch of extraneous `h2` comments. Maybe [mastering markdown](https://guides.github.com/features/mastering-markdown/) && [stackedit](https://stackedit.io) would be worth including in the contributing docs?

File: docs/DCO.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
+Developer Certificate of Origin
```
Comment (wenzowski) [inline docs/DCO.md]:
```suggestion
[Developer Certificate of Origin](https://developercertificate.org/)
```

File: docs/CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
+# H2 How to Contribute:
```
Comment (wenzowski) [inline docs/CONTRIBUTING.md]:
```suggestion
## How to Contribute:
```

Then number of `#` characters determines the header level. Check out [ATX Headings in the spec](https://github.github.com/gfm/#atx-headings) for examples.

File: docs/CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+By contributing to this project you agree to the [Certificate of Origin (DCO)](./DCO.md). This document was created by the Linux Kernel community and is a simple statement that you, as a contributor, have the legal right to make the contribution.
+
+# H2 Support Channels
+
+Before opening a new issue, it's helpful to search the project - it's likely that another user has already reported the issue you're facing, or it's a known issue that we're already aware of.
+[Create An Issue](https://github.com/bcgov/cas-ciip-portal/issues)
+# H3 Getting Started
+  * Fork the repository on GitHub
+  * Read the [documentation](https://github.com/bcgov/cas-ciip-portal/blob/develop/documentation/developer-guide.md) for build instructions
```
Comment (wenzowski) [inline docs/CONTRIBUTING.md]:
```suggestion
  * Read the [developer documentation](./developer-guide.md) for build instructions
```
Line 7 uses a relative reference...sweet! This should too.

File: docs/DCO.md (hunk 5 above)
Comment (wenzowski) [reply docs/DCO.md]: @blueQcloud did you have an opinion on this one?

Comment (blueQcloud) [reply docs/DCO.md]: hmm, you said you wanted to be consistent with the links?

**Pull request merged**